### PR TITLE
Favorites UI

### PR DESCRIPTION
**Pull request by lihter**
Made `FavoritesVC`, `FavoritesDelegate` and `FavoritesPresenter`.
Displayed mocked data.
Not sure how to handle incorrect spacing between the cells.
In one of the next PRs I will deal with the heart icon on movie cells (the heart icon will be filled with white colour), for now I will leave it as it is.

https://user-images.githubusercontent.com/82365531/128733988-dca9d9f9-d5be-4205-837d-fb70d10c47b2.mp4

### CODE REVIEW

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController+Design.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        flowLayout.minimumInteritemSpacing = 2 * offset
+        
+        collectionView.backgroundColor = .clear
+        collectionView.showsVerticalScrollIndicator = false
+    }
+    
+    func defineLayoutForViews() {
+        favouritesLabel.snp.makeConstraints {
+            $0.top.equalToSuperview().offset(9 * offset)
+            $0.leading.equalToSuperview().offset(4 * offset)
```
Comment (marinzg) [inline MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController+Design.swift:40]: Missing `trailing`.

**Review by marinzg — CHANGES_REQUESTED** [review]

**Comment by marinzg** [comment]
Also, the size of the cells is too big to fit 3 of them on the screen and that's why you have the spacing.
You can create custom layout for this, or dynamically calculate cell size:
```swift
    func collectionView(
        _ collectionView: UICollectionView,
        layout collectionViewLayout: UICollectionViewLayout,
        sizeForItemAt indexPath: IndexPath
    ) -> CGSize {
        let numberOfCellsInRow: CGFloat = 3
        let margin = 4 * offset
        let spacing = 4 * offset
        let oneCellWidth = (view.bounds.width - 2 * margin - spacing) / numberOfCellsInRow
        let height = NewMovieCell.cellSize.height / NewMovieCell.cellSize.width * oneCellWidth
        return CGSize(width: oneCellWidth, height: height)
    }
```

File: MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController+Design.swift (hunk 1 above)
Comment (lihter) [reply MovieApp/MovieApp/App/Features/MovieBrowser/MoviesViewController/Favourites/FavoritesViewController+Design.swift:40]: [Here](https://github.com/lihter/Movie-App/pull/37/commits/958114949a4592508dcd2937baff705080c2e9cc#diff-424812fd3fd311a19a5f6478b1e78956455a28ee0a42d9015e2be0950ab131ebR41)

**Comment by lihter** [comment]
Used your suggestion to assign cell size dynamically.

simulator_screenshot_6638B2EA-FE71-4C14-A576-B3D22B6FB3EE

**Review by marinzg — APPROVED** [review]

**Pull request merged**